openmc/deplete/chain.py

```diff
@@ -0,0 +1,786 @@
+                missing_reaction.add(parent)
+                continue
+
+            this_sum = sum(sub.values())
+            # sum of branching ratios can be lower than 1 if no ground
+            # target is given, but never greater
+            if (this_sum >= 1 + tolerance or (grounds[parent] in sub
+                                              and this_sum <= 1 - tolerance)):
+                if strict:
+                    msg = ("Sum of {} branching ratios for {} "
+                           "({:7.3f}) outside tolerance of 1 +/- "
+                           "{:5.3e}".format(
+                               reaction, parent, this_sum, tolerance))
+                    raise ValueError(msg)
+                bad_sums[parent] = this_sum
+            else:
+                rxn_ix_map[parent] = indexes
+                sums[parent] = this_sum
+
+        if len(rxn_ix_map) == 0:
+            raise IndexError(
+                "No {} reactions found in this {}".format(
+                    reaction, self.__class__.__name__))
+
+        if len(missing_parents) > 0:
+            warn("The following nuclides were not found in {}: {}".format(
+                 self.__class__.__name__, ", ".join(sorted(missing_parents))))
+
+        if len(missing_reaction) > 0:
+            warn("The following nuclides did not have {} reactions: "
+                 "{}".format(reaction, ", ".join(sorted(missing_reaction))))
+
+        if len(missing_products) > 0:
+            tail = ("{} -> {}".format(k, v)
+                    for k, v in sorted(missing_products.items()))
+            warn("The following products were not found in the {} and "
+                 "parents were unmodified: \n{}".format(
+                     self.__class__.__name__, ", ".join(tail)))
+
+        if len(bad_sums) > 0:
+            tail = ("{}: {:5.3f}".format(k, s)
+                    for k, s in sorted(bad_sums.items()))
+            warn("The following parent nuclides were given {} branch ratios "
+                 "with a sum outside tolerance of 1 +/- {:5.3e}:\n{}".format(
+                     reaction, tolerance, "\n".join(tail)))
+
+        # Insert new ReactionTuples with updated branch ratios
+
+        for parent_name, rxn_index in rxn_ix_map.items():
+
+            parent = self[parent_name]
+            new_ratios = branch_ratios[parent_name]
+            rxn_index = rxn_ix_map[parent_name]
+
+            # Assume Q value is independent of target state
+            rxn_Q = parent.reactions[rxn_index[0]].Q
+
+            # Remove existing reactions
+
+            for ix in reversed(rxn_index):
+                parent.reactions.pop(ix)
+
+            all_meta = True
+
+            for tgt, br in new_ratios.items():
+                all_meta = all_meta and ("_m" in tgt)
+                parent.reactions.append(ReactionTuple(
+                    reaction, tgt, rxn_Q, br))
+
+            if all_meta and sums[parent_name] != 1.0:
```

Comment:
Optional: I would add a comment explaining the expected behavior of this block.  By my read, it has to do with expected appended reactions related to when / if unity happens for a sum, however it's not 100% clear from context.  Thanks!